### PR TITLE
Transcripts - Tracks

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
This PR adds Track events for transcripts.

## Testing Instructions
** `player_shelf_action_tapped` event is tested in https://github.com/Automattic/pocket-casts-android/pull/2512
** Below events are tracked with podcast and episode uuids as properties

1. Play an episode having transcript
2. Tap on the player shelf transcript button
3. Notice that `transcript_shown` is tracked 
4. Pull to refresh
5. Notice that `transcript_pulled_to_refresh` is tracked 
6. Go back either by tapping close button or by swipe back gesture
7. Notice that `transcript_dismissed` is tracked 
8. Play another episode with a transcript that you haven't opened before
9. Disable internet
10. Tap transcript shelf icon on the player
11. Notice that an error message is shown and `transcript_error` is tracked
12. Enable internet and tap "Try again"
13. Notice that transcript is shown and `transcript_shown` is tracked 
14. Tap search button
15. Notice `transcript_search` is tracked
16. Search for a term and tap next button in the search bar
17.  Notice `transcript_search_next_result` is tracked
18. Tap previous button in the search bar
19. Notice that `transcript_search_previous_result` is tracked

## Checklist
- [x] ~~If this is a user-facing change, I have added an entry in CHANGELOG.md~~
- [X] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] ~~I have considered whether it makes sense to add tests for my changes~~
- [x] ~~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~~
- [x] ~~Any jetpack compose components I added or changed are covered by compose previews~~
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

#### I have tested any UI changes...

- [x] ~~with different themes~~
- [x] ~~with a landscape orientation~~
- [x] ~~with the device set to have a large display and font size~~
- [x] ~~for accessibility with TalkBack~~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mebarbosa — APPROVED** [review]
All events worked as expected!

@ashiagr  Should we add an event for the clear button on search? 
image

File: modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptSearchViewModel.kt [hunk 1]
```diff
@@ -66,18 +75,24 @@ class TranscriptSearchViewModel @Inject constructor(
         }
     }
 
+    fun onSearchButtonClicked() {
+        track(AnalyticsEvent.TRANSCRIPT_SEARCH)
```
[the remark this answers is not in the document]
Comment (SergioEstevao) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptSearchViewModel.kt:79]: Looking at other search related events and maybe `TRANSCRIPT_SEARCH_BUTTON_TAPPED`, it looks `SHOW` events are more used when we display a new screen. What do you think @leandroalonso  ?

[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptSearchViewModel.kt]: Renamed it to `TRANSCRIPT_SEARCH_SHOWN` in cd89a3f.

**Comment by ashiagr** [comment]
> Should we add an event for the clear button on search

I checked it on other search screens and noticed we don't track this event. I think it is fine to not track it.

File: modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptSearchViewModel.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (leandroalonso) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptSearchViewModel.kt]: I'm confused, did we change the naming of any event?

**Pull request merged**